modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskActivityServiceImpl.java

```diff
@@ -69,18 +70,22 @@ public boolean addSuccessfulExecution(Long activityId) {
 
         TaskExecutionProgress progress = activity.getTaskExecutionProgress();
         progress.addSuccess();
-        boolean taskFinished = progress.getActionsSucceeded() == progress.getTotalActions();
 
-        if (taskFinished) {
-            activity.setActivityType(TaskActivityType.SUCCESS);
-            activity.setMessage(TASK_SUCCEEDED);
-            activity.getFields().clear();
+        return updateTaskProgress(progress, activity);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public void addFilteredExecution(Long activityId) {
+        TaskActivity activity = taskActivitiesDataService.findById(activityId);
+        if (activity == null) {
+            return;
```

Comment:
omit using return in void methods, it is less readable and it is hard to debugging